resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/k8s/ExecutorPodsSnapshot.scala

```diff
@@ -42,32 +43,47 @@ object ExecutorPodsSnapshot extends Logging {
     ExecutorPodsSnapshot(toStatesByExecutorId(executorPods))
   }
 
-  def apply(): ExecutorPodsSnapshot = ExecutorPodsSnapshot(Map.empty[Long, ExecutorPodState])
+  def apply(): ExecutorPodsSnapshot = ExecutorPodsSnapshot(Map.empty[Long, ExecutorState])
 
-  private def toStatesByExecutorId(executorPods: Seq[Pod]): Map[Long, ExecutorPodState] = {
+  private def toStatesByExecutorId(executorPods: Seq[Pod]): Map[Long, ExecutorState] = {
     executorPods.map { pod =>
       (pod.getMetadata.getLabels.get(SPARK_EXECUTOR_ID_LABEL).toLong, toState(pod))
     }.toMap
   }
 
-  private def toState(pod: Pod): ExecutorPodState = {
+  private def toState(pod: Pod): ExecutorState = {
     if (isDeleted(pod)) {
-      PodDeleted(pod)
+      ExecutorPodDeleted(pod)
     } else {
       val phase = pod.getStatus.getPhase.toLowerCase(Locale.ROOT)
       phase match {
         case "pending" =>
-          PodPending(pod)
+          ExecutorPending(pod)
         case "running" =>
-          PodRunning(pod)
+          // Checking executor container status is not terminated
```

Comment:
Can we add a test for this?

Reply:
Is the failedExecutorWithRuningSidecar bellow now what you're looking for?